pixeltable/store.py

```diff
@@ -243,6 +247,25 @@ def insert_rows(
                     table_rows: List[Dict[str, Any]] = []
                     for row_idx in range(batch_start_idx, min(batch_start_idx + batch_size, len(row_batch))):
                         row = row_batch[row_idx]
+                        for slot_idx in row.img_slot_idxs:
+                            if row.file_paths[slot_idx] is not None:
+                                try:
+                                    PIL.Image.open(row.file_paths[slot_idx])
+                                except Exception as e:
+                                    # row.vals[slot_idx] = None
+                                    row.has_val[slot_idx] = False
```

Comment:
This needs to happen when the slot in the datarow gets set. That way, you can take that into account and avoid evaluating exprs on that slot.